history.go

```diff
@@ -0,0 +1,56 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+)
+
+var (
+	ErrHistoryExisted = errors.New("[merkletree] trying to override existing history")
+	ErrBadEpoch       = errors.New("[merkletree] next epoch's STR has bad epoch ")
+)
+
+var currentSTR *SignedTreeRoot
```

Comment:
It would be nice to eliminate this global and create a `History` struct, like

    type History struct {
       m *MerkleTree
       currentSTR *SignedTreeRoot
       epochInterval int64
    } 

then, instead of `InitHistory`, you can have,

    func NewHistory(m *MerkleTree, startEp, epInterval int64) *History {
       h := new(History)
       h.m = m
       h.epochInterval = epInterval
       h.currentSTR = m.generateSTR(startEp, 0, make([]byte, crypto.HashSizeByte))
       return h
    }

something like that. Then the rest of these methods can have the signature `func (h *History)`. How does that sound?